File: src/neoforge/game/common/main/java/settingdust/preloading_tricks/neoforge/util/LoaderAdapter.java
package settingdust.preloading_tricks.neoforge.util;

import net.neoforged.fml.loading.LoadingModList;
import settingdust.preloading_tricks.util.LoaderAdapter;

public class LoaderAdapter implements LoaderAdapter {
    @Override
    public boolean isClient() {
        return NeoForgeAdapter.getInstance().getDist().isClient();
    }

    @Override
    public java.nio.file.Path gameDir() {
        return net.neoforged.fml.loading.FMLPaths.GAMEDIR.get();
    }

    @Override
    public boolean isModLoaded(String modId) {
        return LoadingModList.get().getModFileById(modId) != null;
    }
}

File: src/neoforge/1.21/main/kotlin/settingdust/preloading_tricks/neoforge/util/LoaderAdapter.kt
package settingdust.preloading_tricks.neoforge.util

import net.neoforged.fml.loading.FMLLoader
import net.neoforged.fml.loading.LoadingModList
import settingdust.preloading_tricks.util.LoaderAdapter

class LoaderAdapter : LoaderAdapter {
    override val isClient: Boolean
        get() = FMLLoader.getDist().isClient

    override fun isModLoaded(modId: String) = LoadingModList.get().getModFileById(modId) != null
}